compiler/parse_aux.ml

```diff
@@ -164,11 +164,11 @@ let rec apply_type_attribute ty attr =
   | (("managed", _), Type_bigarray(attrs, ty_elt)) ->
       Type_bigarray({attrs with malloced = true}, ty_elt)
   | (("switch_is", [rexp]), Type_union(name, attr)) ->
-      Type_union(name, {attr with discriminant = rexp})
+      Type_union(name, {discriminant = rexp})
   | (("switch_is", [rexp]), Type_pointer(attr, Type_union(name, attr'))) ->
```

Comment:
Warning be damned, I prefer the original formulation, `{attrs with ... }`.  It doesn't matter that currently there is only one field in those record types.  (One day there might be several.)  What matters is that the attribute modifies the corresponding field and nothing else.